core/src/main/kotlin/me/fornever/todosaurus/core/issues/ToDoService.kt

```diff
@@ -41,7 +41,12 @@ class ToDoService(private val project: Project, private val scope: CoroutineScop
                 .getChoiceOrNull()
 
             if (savedChoice != null) {
-                val model = retrieveWizardContextBasedOnUserChoice(toDoItem, savedChoice)
+                val model = try {
+                    retrieveWizardContextBasedOnUserChoice(toDoItem, savedChoice)
+                }
+                catch (exception: Exception) {
+                    return@launch Notifications.Memoization.failed(exception, toDoItem, project)
+                }
```

Comment:
There are two issues with this block.

1. Almost any block is expected to pass-through cancellation exceptions.
2. In JVM, you better catch `Throwable` and not just `Exception`.

So let's rewrite this as
```
catch (t: Throwable) {
  if (t is CancellationException) throw t
  return@launch Notifications.Memoization.failed(exception, toDoItem, project)
}
```

Reply:
Am I understanding correctly that we will need to rewrite all `try-catch` blocks in `Todosaurus` using `Throwable`?

Reply:
Yep, preferably. But don't forget about cancellation.